R/TuningInstanceMulticrit.R

```diff
@@ -41,9 +41,19 @@ TuningInstanceMultiCrit = R6Class("TuningInstanceMultiCrit",
     #' This defines the resampled performance of a learner on a task, a
     #' feasibility region for the parameters the tuner is supposed to optimize,
     #' and a termination criterion.
-    initialize = function(task, learner, resampling, measures, search_space,
-      terminator, store_models = FALSE, check_values = FALSE,
-      store_benchmark_result = TRUE) {
+    initialize = function(task, learner, resampling, measures,
+      terminator, search_space = NULL, store_models = FALSE,
+      check_values = FALSE, store_benchmark_result = TRUE) {
+      learner = assert_learner(as_learner(learner, clone = TRUE))
+
+      if (!is.null(search_space) && length(learner$param_set$get_values(type = "only_token")) > 0) {
+        stop("If the values of the ParamSet of the Learner contain TuneTokens you cannot supply a search_space.")
+      }
+      if (is.null(search_space)) {
+        search_space = learner$param_set$search_space()
+        learner$param_set$values = learner$param_set$get_values(type = "without_token")
```

Comment:
don't do this here, gives a problem when there are requirements. instead do it in the evaluation code in `ObjectiveTuning$eval_many` like [here](https://github.com/mlr-org/mlr3tuning/pull/282/files#diff-6414282b6766410af11f1d8bee6f003ddac07ced005f6a71a24142eae6f7addcR82)

Reply:
Wait, if this is not possible here how was it even possible before we introduced the tune_tokens? It was'nt right? So your example is the following:

We tune over Param A which can have values aX and aY.
We set Param X that needs A to be aX in the `param_set$values` to a fixed value - as well we set Param Y that needs A to ba aY to a fixed value.... right?

This was not possible before, and should be possible now?

Reply:
```
pipeop = po("branch", options = c("kknn", "svm", "ranger"))

pipeop$param_set

<ParamSet:branch>
          id    class lower upper          levels        default value
1: selection ParamFct    NA    NA kknn,svm,ranger <NoDefault[3]>  kknn
```

`selection` has the tag `required`. `PipeOpBranch` automatically sets the required value to the first option. Currently we try to remove this values / `TuneToken` in `TuningInstanceSingleCrit`.

Reply:
This was possible before because we just didn't touch the `param_set$values` before, and in the tuning loop we just overwrite `param_set$values` without clearing values.